src/net.rs

```diff
@@ -147,7 +149,7 @@ const fn is_unicast_link_local(addr: Ipv6Addr) -> bool {
 }
 
 /// Given a listen/bind address, finds all the local addresses for that address family.
-pub fn find_local_addresses(listen_addr: SocketAddr) -> Vec<SocketAddr> {
+pub(crate) fn find_local_addresses(listen_addr: SocketAddr) -> Result<Vec<SocketAddr>> {
```

Comment:
I don't think this should be pub, and probably should even be in the provider module.  Though that would make the diff harder to read.